ember-cli-flash/src/services/flash-messages.js

```diff
@@ -48,30 +46,28 @@ export default class FlashMessagesService extends Service {
   }
 
   clearMessages() {
-    const flashes = this.queue;
-
-    if (isNone(flashes)) {
+    if (isNone(this.queue)) {
       return;
     }
 
-    flashes.forEach((flash) => flash.destroyMessage());
-    flashes.clear();
+    this.queue.forEach((flash) => flash.destroyMessage());
+    this.queue = [];
 
     return this;
   }
 
-  registerTypes(types = emberArray()) {
+  registerTypes(types = []) {
     types.forEach((type) => this._registerType(type));
 
     return this;
   }
 
   peekFirst() {
-    return this.queue.firstObject;
+    return this.queue.at(0);
   }
 
   peekLast() {
-    return this.queue.lastObject;
+    return this.queue.at(-1);
```

Comment:
Just in case someone has to support old Safari: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/at#browser_compatibility

Reply:
Interesting, I was wondering about this

 In a v1 addon would this have been transpiled by babel?

Reply:
only if configured to use some polyfill